umich_api/api_utils.py

```diff
@@ -141,6 +172,12 @@ def get_access_token(self, token_url, client_scope):
         resp = requests.post(f"{self.base_url}/{token_url}", data=payload, headers=headers)
         try:
             if (resp.ok):
-                return resp.json().get('access_token')
+                r_json = resp.json()
+                # If expires_at is a parameter on the return, add the value to "now" and store it as a new value expires_time
+                if 'expires_at' in r_json:
+                    r_json['expires_time'] = datetime.now() + timedelta(seconds=r_json['expires_at'])
+                # Cache the token
+                self.tokens[token_url] = r_json
+                return r_json
         except (ValueError):
             self.__log.error ("Error obtaining access token with credentials")
```

Comment:
more context should be added which scope token generation failed like `Error obtaining an access token for {client_scope} {token_url}

Reply:
```suggestion
            self.__log.error (f"Error obtaining access token for {client_scope} {token_url}")
```